File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n, m = len(matrix), len(matrix[0])
        list_indices = []
        
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    list_indices.append((i,j))
  
        for pair in list_indices:
            i, j = pair
            matrix[i] = [0]*m
            for row in range(n):
                matrix[row][j] = 0
        
        return matrix

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:

	void setZeroes(std::vector<std::vector<int>>& matrix) {

		int n = matrix.size(), m = matrix[0].size();

		std::vector<std::vector<int>> indices = {};

		for (int i = 0; i < n; i++) {

			for (int j = 0; j < m; j++) {

				if (matrix[i][j] == 0) {
					indices.push_back({ i,j });
				}


			}


		}


		for (int idx = 0; idx < indices.size(); idx++) {
			int i = indices[idx][0], j = indices[idx][1];

			matrix[i] = std::vector<int>(m, 0);
			for (int row = 0; row < n; row++) {
				
                if(row != i){matrix[row][j] = 0;}
			}

		}


	}




};